### ISSUE TITLE
BEHRT: Transformer for Electronic Health Records (1)

### ISSUE THREAD

**Issue by ghuijo** [yskim0/One-Day-One-Paper#160]
## Abstract & Introduction
* BEHRT, A deep neural sequence transduction model for electronic health records 제안.
* personalised interpretation of its predictions을 가능하게 함.
* flexible architecture를 가짐. --> 발전 & 활용 가능성.
* EHR 특징
  * contain “concepts” such as diagnoses, interventions, lab tests, clinical narratives
  * a single or multiple data points
  * relies on experts’ ability

## Related Study (Intro 중에서)
* DL model that can predict the risk of lung cancer from a patient’s tomography images
* DL can predict a range of cardiovascular risk factors from just a retinal fundus photograph
* providing classifiers with useful patient representations to predict the onset of a number of diseases from EHR
* a convolutional neural network (CNN) model called Deepr (Deep record) for predicting the probability of readmission
* a shallow recurrent neural network (RNN) model to predict the diagnoses and medications that are likely to occur in the subsequent visit
* Long Short-Term Memory (LSTM) architecture with attention mechanism, called DeepCare, which outperformed standard ML techniques, plain LSTM, and plain RNN in tasks such as prediction of the onset of diabetes
*  reverse-time attention mechanism to consolidate past influential visits using an end-to-end RNN model named RETAIN for the prediction of heart failure

#### Data
*  Clinical Practice Research Datalink (CPRD)
* 그 중에서도 GP (general practitioner)
* 35 million patients
* only considered the data from GP practices that consented to (and hence have) record linkage with HES
  * --> hospital episode statistics

#### Pre-processing of CPRD
* start with 8 million patients & at least 5 visits 
* total of =G 301 codes for diagnoses
* 데이터 준비 과정
  <image src="https://user-images.githubusercontent.com/67627471/134675855-b769cb0c-592e-4d41-a89a-1a1e161f821b.png" width="80%" height="80%" />
  <image src="https://user-images.githubusercontent.com/67627471/134675889-0acc5914-3595-4635-ac8c-0c3bad439514.png" width="80%" height="80%" />

#### BEHRT: A Transformer-based Model for EHR
* aim to use a given patient’s past EHR to predict his/her future diagnoses (if any), as a multi-label classification problem
* four key challenges
  * complex and nonlinear interactions among past, present and future concepts
  * long-term dependencies among concepts (e.g., diseases occurring early in the history of a patient effecting events far later in future)
  * difficulties of representing multiple heterogeneous concepts of variable sizes and forms to the model
  * the irregular intervals between consecutive visits
* BEHRT 구조.
  <image src="https://user-images.githubusercontent.com/67627471/134676258-b6e078a2-564b-40a9-984d-293585e146db.png" width="80%" height="80%" />
* diagnoses는 단어로, 각 visit는 문장으로, a patient’s entire medical history는 문서로 보고 multi-head self-attention, positional encoding, masked language model (MLM) 수행함.
* pre-train deep bidirectional representations of medical concepts by jointly conditioning on both left and right contexts in all layers
* feed-forward neural networks to model the temporal evolution of EHR data through utilising various forms of sequential concepts in the data
  * alleviates the exploding and vanishing gradient problems and captures information by simultaneously considering the full sequence
  * more efficient training through learning the data in parallel rather than in sequence (unlike the RNN)
* embedding layer
  *  four embeddings: disease, position, age, visit segment
  * Positional encodings determine the relative position of concepts in EHR sequence, and enable the network to capture the positional interactions among diseases
  *  pre-determined encoding 사용
  * --> avoid weak learning of positional embedding caused by imbalanced distribution of medical sequence length in EHR
  * embedding age and linking it to each visit/diagnosis
  * --> provide the model with a sense of time &inform it of age as a universal epidemiological notion of when things happened
  * Visit segment --> providing BEHRT with extra information to indicate the separation between visits
  * attention mechanism purely investigates intra-visit relationships among diseases

#### Pre-training BEHRT using masked language model (MLM)
* deep bidirectional model is more powerful than either a left-to-right model or the shallow concatenation of a left-to-right and a right-to-left mode
* initialized disease, age, and segment embeddings randomly, and the positional encoding
* 1. left 86.5% of the disease words unchanged
* 2. 12% of the words were replaced with [mask]
* 3. remaining 1.5% of words were replaced with randomly-chosen disease words
* --> the replacement of the disease words acts as injected noise into the model

#### Disease Prediction
* 3가지 예측 수행
  * prediction of diseases in the next visit (T1)
  * prediction of diseases in the next 6 months (T2)
  * prediction of diseases in the next 12 months
* randomly allocated the patients into two groups: train and test
* randomly choose an index j ( 3 < j < p) for each patient
* feed these input medical histories into BEHRT for feature extraction
* the network pools the information into a representation of the patient
* passes it along to a single feed-forward classifier layer for output
* --> 여기서 아웃풋은 subsequent visit prediction
* 평가 매트릭
  * AUROC / APS를 환자별로 구해서 모든 환자마다 두 값을 평균냄.

##
* EHR을 sentence로 보고 transformer를 적용한다는 놀라운 아이디어.
* 학습에 사용한 데이터도 크고, pre-trained 모델로 활용될 여지가 충분한 듯.
* 이걸로 논문 리뷰 발표해야겠음.
